share/spice/detect_lang/detect_lang.js

```diff
@@ -1,149 +1,153 @@
-function ddg_spice_detect_lang (api_result) {
+(function(env) {
     "use strict";
+    env.ddg_spice_detect_lang = function (api_result) {
 
-    // Check for any errors.
-    if(!api_result || !api_result.data || !api_result.data.detections || api_result.data.detections.length === 0) {
-        return Spice.failed('detect_lang');
-    }
+        // Check for any errors.
+        if(!api_result || !api_result.data || !api_result.data.detections || api_result.data.detections.length === 0) {
+            return Spice.failed('detect_lang');
+        }
 
-    var query = "";
-    $("script").each(function() {
-        var matched, result;
-        matched = $(this).attr("src");
-        if(matched) {
-            result = matched.match(/\/js\/spice\/detect_lang\/([^\/]+)/);
-            if(result) {
-                query = decodeURIComponent(result[1]);
+        var query = "";
+        $("script").each(function() {
```

Comment:
We now have a slighter better way of doing this:

https://github.com/duckduckgo/zeroclickinfo-spice/blob/e535f4bbfdae58990f33d64c9d7bde9bf4991038/share/spice/expand_url/expand_url.js#L6